runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/fn/control/MSecMonitoringInfoToCounterUpdateTransformer.java

```diff
@@ -120,14 +110,19 @@ public CounterUpdate transform(MonitoringInfo monitoringInfo) {
       return null;
     }
 
-    long value = monitoringInfo.getMetric().getCounterData().getInt64Value();
+    long value;
+    try {
+      value = VARINT_CODER.decode(monitoringInfo.getPayload().newInput());
```

Comment:
Ditto here, please use a one liner helper method to extract the value, here and in all the transformer classes.